### PR TITLE
docs: Add documentation for Zitadel SSO integration

### PR DESCRIPTION
**Pull request by stephanrenggli**
Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [x] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)). 
* [ ] My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.
* [ ] I have added a brief description of why this PR is necessary and/or what this PR solves.

Please see [Contribution FAQs](https://argo-cd.readthedocs.io/en/latest/developer-guide/faq/) if you have questions about your pull-request.

This PR adds documentation to the User Management section to document integration of Zitadel SSO with ArgoCD as proposed in this discussion: https://github.com/argoproj/argo-cd/discussions/11855#

### CODE REVIEW

File: docs/operator-manual/user-management/zitadel.md [hunk 1]
```diff
@@ -0,0 +1,210 @@
+# Zitadel
+Please also consult the [Zitadel Documentation](https://zitadel.com/docs).
+## Integrating Zitadel and ArgoCD
+These instructions will take you through the entire process of getting your ArgoCD application authenticating with Zitadel. You will create an application within Zitadel and configure ArgoCD to use Zitadel for authentication using roles set in Zitadel to determine privileges in ArgoCD.
```
Comment (fforootd) [inline docs/operator-manual/user-management/zitadel.md:4]:
```suggestion
These instructions will take you through the entire process of getting your ArgoCD application authenticating and authorizing with Zitadel. You will create an application within Zitadel and configure ArgoCD to use Zitadel for authentication using roles set in Zitadel to determine privileges in ArgoCD.
```

File: docs/operator-manual/user-management/zitadel.md [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+
+The following steps are required to integrate ArgoCD with Zitadel:
+1. Create a new project and a new application in Zitadel
+2. Configure the application in Zitadel
+3. Set up roles in Zitadel
+4. Set up a action in Zitadel
```
Comment (fforootd) [inline docs/operator-manual/user-management/zitadel.md:10]:
```suggestion
4. Set up an action in Zitadel
```

**Review by fforootd — COMMENTED** [review]
Nice PR!

Only some really small feedbacks from my end.

Thank you for doing this.

**Review by fforootd — APPROVED** [review]

File: docs/operator-manual/user-management/zitadel.md [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+- Zitadel Application: `argocd-application`
+- Zitadel Action: `groupsClaim`
+- ArgoCD FQDN: `argocd.example.com`
+- ArgoCD Administrator Role: `argocd_administrators`
+- ArgoCD User Role: `argocd_users`
+
+You may choose different values in your setup; these are used to keep the guide consistent.
+
+## Setting up your project and application in Zitadel
+First we will create a new project within Zitadel. Go to **Projects** and select **Create New Project**.
```
Comment (todaywasawesome) [inline docs/operator-manual/user-management/zitadel.md:26]:
```suggestion
First, we will create a new project within Zitadel. Go to **Projects** and select **Create New Project**.  
```

File: docs/operator-manual/user-management/zitadel.md [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+    g, argocd_administrators, role:admin
+    g, argocd_users, role:readonly
+  policy.default: ''
+```
+
+The roles specified under `policy.csv` must match the roles configured in Zitadel.  
+The Zitadel role `argocd_administrators` will be assigned the ArgoCD role `admin` granting admin access to ArgoCD.  
+The Zitadel role `argocd_users` will be assigned the ArgoCD role `readonly` granting read-only access to ArgoCD.
+
+Deploy your ArgoCD configmaps. ArgoCD and Zitadel should now be setup correctly to allow users to log in to ArgoCD using Zitadel.
```
Comment (todaywasawesome) [inline docs/operator-manual/user-management/zitadel.md:200]:
```suggestion
Deploy your ArgoCD configmaps. ArgoCD and Zitadel should now be set up correctly to allow users to log in to ArgoCD using Zitadel.
```

**Review by todaywasawesome — CHANGES_REQUESTED** [review]
Looks great, I made a few small grammar edits. Once those are accepted, I'll approve.

**Comment by stephanrenggli** [comment]
Thanks for the suggestions!

**Review by todaywasawesome — APPROVED** [review]

**Comment by jannfis** [comment]
Thanks! Unfortunately, this page is not linked from the navigation menu. It needs to be linked [here](https://github.com/argoproj/argo-cd/blob/dc8d729078df6fe9d4c480df0e67b2402af75e0b/mkdocs.yml#L29)

**Review by pasha-codefresh — APPROVED** [review]

**Pull request merged**